compilers/concrete-compiler/compiler/lib/Dialect/FHELinalg/IR/FHELinalgOps.cpp

```diff
@@ -1687,6 +1687,488 @@ void ToUnsignedOp::getCanonicalizationPatterns(
   getSignedConvCanonicalizationPatterns<ToUnsignedOp>(patterns, context);
 }
 
+std::optional<mlir::Value>
+fuseBackToBackTableLookups(mlir::Value value, mlir::PatternRewriter &rewriter) {
+
+  using mlir::concretelang::FHE::FheIntegerInterface;
+
+  auto currentOperation = value.getDefiningOp();
+
+  auto currentOperationIsTlu =
+      llvm::isa<ApplyLookupTableEintOp>(currentOperation);
```

Comment:
You can avoid the invocations of `isa` by doing directly a `dyn_cast` and using the resulting values as boolean values in line 1702 and as-is in lines 1708 and 1709.